Reworked plugin queues (#5088)
Kong/docs.konghq.com: app/_includes/md/plugins-hub/queue-parameters.md

<!-- shared with plugins that use queues. -->
    - name: queue.max_batch_size
      type: number
      default: 1
      description: Maximum number of entries to be processed together
          as a batch.
    - name: queue.max_coalescing_delay
      type: number
      default: 1
      description: >
          Maximum number of seconds (as a fraction) that can elapse
          after the first entry was queued before the queue starts
          processing entries.  This parameter has no effect when
          `batch_max_size` is one because queued entries will be sent
          immediately in that case.
    - name: queue.max_entries
      type: number
      default: 10000
      description: >
          Maximum number of entries that can be waiting on the queue.
          Once this number of requests is reached, the oldest entry is
          deleted from the queue before a new one is added.
    - name: queue.max_bytes
      type: number
      default: nil
      description:
          Maximum number of bytes that can be waiting on a queue.
          Once this many bytes are present on a queue, old entries
          up to the size of a new entry to be enqueued are deleted

>           from the queue.

Reviewer: ```suggestion
          from the queue.
          minimum_version: "3.3.x"
```

    - name: queue.max_retry_time
      type: number
      default: 60
      description: >
          Time (in seconds) before the queue gives up trying to send a
          batch of entries.  Once this time is exceeded for a batch,
          it is deleted from the queue without being sent.  If
          this parameter is set to -1, no retries will be made for a

>           failed batch.

Reviewer: ```suggestion
          failed batch.
          minimum_version: "3.3.x"
```

    - name: queue.max_retry_delay
      type: number
      default: 60
      description: >
          Maximum time (in seconds) between retries sending a batch of
          entries. The interval between retries follows an
          exponential back-off algorithm capped at this number of

>           seconds.

Reviewer: ```suggestion
          seconds.
          minimum_version: "3.3.x"
```

<!-- shared with plugins that use queues. -->

This plugin utilizes a queue to decouple the production and
consumption of data.  This reduces the number of concurrent requests
made to the upstream server under high load situations and provides
for buffering during temporary network or upstream outages.

> A number of parameters can be tuned to influence the behavior and
> capacity of the queues used by the plugin.  Please see [XXX](the
> documentation on queues) for detail information on the usage of these
> parameters.

Reviewer: ```suggestion
You can set several parameters to configure the behavior and
capacity of the queues used by the plugin. For more information about how to use these parameters, see [Batch Queuing Reference](gateway/latest/kong-plugins/batch-queue/reference/) in the {{site.base_gateway}} documentation.
```
I think that's the link you were looking for?

The queue parameters all reside in a record under the key `queue` in
the `config` parameter section of the plugin.

> parameters:

Reviewer: Did you want this line and below to display like the table like the rest of the parameters? I think it would have to live in the same spot as those other parameters (I don't think it will render in this section).

Author: Can we somehow make the queue parameters into a separate table for each plugin?

Reviewer: @hanshuebner I asked about making the parameters into a separate table and it's not possible. The queue parameters have to be included in each plugin’s index file manually as part of the params section.
